lib/modules/datasource/nextcloud/index.ts

```diff
@@ -0,0 +1,81 @@
+import { cache } from '../../../util/cache/package/decorator';
+import { regEx } from '../../../util/regex';
+import { asTimestamp } from '../../../util/timestamp';
+import * as semver from '../../versioning/semver';
+import { Datasource } from '../datasource';
+import type { GetReleasesConfig, ReleaseResult } from '../types';
+import { Applications } from './schema';
+
+export class NextcloudDatasource extends Datasource {
+  static readonly id = 'nextcloud';
+
+  private static readonly defaultTranslationLanguage = 'en';
+
+  private static readonly sourceUrlRegex = regEx(
+    /(?<prefix>.*github.com\/nextcloud)(?<suffix>\/.*)/,
+  );
+
+  override readonly defaultVersioning = semver.id;
+
+  constructor() {
+    super(NextcloudDatasource.id);
+  }
+
+  @cache({
+    namespace: `datasource-${NextcloudDatasource.id}`,
+    key: ({ registryUrl, packageName }: GetReleasesConfig) =>
+      `${registryUrl}:${packageName}`,
+  })
+  async getReleases({
+    packageName,
+    registryUrl,
+  }: GetReleasesConfig): Promise<ReleaseResult | null> {
+    if (!registryUrl) {
+      return null;
+    }
+
+    const response = await this.http.getJson(registryUrl, Applications);
```

Comment:
Is there no endpoint to directly fetch the releases for a particular application?

If no, how big can the current JSON response get?